src/lightning_lite/utilities/__init__.py

```diff
@@ -0,0 +1,39 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""General utilities."""
+
+from lightning_lite.utilities.apply_func import move_data_to_device  # noqa: F401
+from lightning_lite.utilities.distributed import AllGatherGrad  # noqa: F401
+from lightning_lite.utilities.enums import _AcceleratorType, _StrategyType, AMPType, LightningEnum  # noqa: F401
+from lightning_lite.utilities.imports import (  # noqa: F401
+    _HIVEMIND_AVAILABLE,
+    _HOROVOD_AVAILABLE,
+    _HPU_AVAILABLE,
+    _IPU_AVAILABLE,
+    _IS_INTERACTIVE,
+    _IS_WINDOWS,
+    _module_available,
+    _POPTORCH_AVAILABLE,
+    _TORCH_GREATER_EQUAL_1_10,
+    _TORCH_GREATER_EQUAL_1_11,
+    _TORCH_GREATER_EQUAL_1_12,
+    _TPU_AVAILABLE,
+    _XLA_AVAILABLE,
+)
```

Comment:
I suggest we don't do this in Lite for non-public things, it's bad for dependency resolution, (also why I want to move them to their files in PL)

With this, I mean specifically importing them in the `__init__.py`

Reply:
Yes, you can

```suggestion
# TODO(lite): Avoid importing protected attributes in `__init__.py` files
from lightning_lite.utilities.imports import (  # noqa: F401
    _HIVEMIND_AVAILABLE,
    _HOROVOD_AVAILABLE,
    _HPU_AVAILABLE,
    _IPU_AVAILABLE,
    _IS_INTERACTIVE,
    _IS_WINDOWS,
    _module_available,
    _POPTORCH_AVAILABLE,
    _TORCH_GREATER_EQUAL_1_10,
    _TORCH_GREATER_EQUAL_1_11,
    _TORCH_GREATER_EQUAL_1_12,
    _TPU_AVAILABLE,
    _XLA_AVAILABLE,
)
```

Reply:
I would also not import protected in the package init...

Reply:
added todo(lite). This will be completed by #11826 and this will minimize all these protected imports. In this temporary state of refactors, we have no other choice